components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -366,34 +367,37 @@ class BrowserToolbar @JvmOverloads constructor(
     /**
      * An action button to be added to the toolbar.
      *
-     * @param imageResource The drawable to be shown.
+     * @param imageDrawable The drawable to be shown.
      * @param contentDescription The content description to use.
      * @param visible Lambda that returns true or false to indicate whether this button should be shown.
      * @param background A custom (stateful) background drawable resource to be used.
      * @param padding a custom [Padding] for this Button.
      * @param listener Callback that will be invoked whenever the button is pressed
      */
     open class Button(
-        imageResource: Int,
+        imageDrawable: Drawable,
         contentDescription: String,
         visible: () -> Boolean = { true },
-        @DrawableRes background: Int? = null,
-        private val padding: Padding = defaultActionPadding,
+        @DrawableRes background: Int = 0,
+        padding: Padding = DEFAULT_PADDING,
         listener: () -> Unit
-    ) : Toolbar.ActionButton(imageResource, contentDescription, visible, background, listener = listener) {
+    ) : Toolbar.ActionButton(imageDrawable, contentDescription, visible, background, padding, listener) {
+
         override fun createView(parent: ViewGroup): View {
-            val view = super.createView(parent)
-            view.setPadding(padding)
-            return view
+            return super.createView(parent).apply {
+                padding?.let {
+                    setPadding(it)
+                }
+            }
```

Comment:
This is very confusing. `padding?` is referencing the property of the parent class. Since you are calling `super.createView()` here.. shouldn't the padding be applied already?

Reply:
Hmm.. I didn't quite look at the reasoning behind why this original code was here when I refactored it. I'll look into it. 👍

Reply:
This was my error in terms of visibility scope. The original change has the super padding as private, and the browser padding as a val so that it couldn't be optional for BrowserToolbar I imagine. I've fixed this now to reflect the original changes.

Reply:
But we are still applying the padding twice, aren't we? 🤔

Reply:
From what I understand, it's to have a default padding only in browser which does look odd. I can file a follow up to fix this.

Reply:
Created https://github.com/mozilla-mobile/android-components/issues/1270